### PR TITLE
Bug 1240554 ui showcase enable facebook fix

### PR DESCRIPTION
**Pull request by chrafuse**
@dmose @Standard8: Added tests for parseStringToElements and formatURL.  Only issue is that shared tests fails for the Unexpected Logged Warnings and Errors Check because parseStringToElements test that I added emits an error.  How do we send true for the suppressConsoleError in the LinkifiedTextView_test to not emit the error -OR- do we remove the test -OR- do we set parseStringToElements call to formatURL to suppressConsoleError(no errors ever emitted)?

### CODE REVIEW

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by chrafuse** [comment]
@dmose @Standard8: Changed to log and trace instead of error.

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by chrafuse** [comment]
Reverted changes to fake data in UI showcase.  Renders the view.

File: shared/js/utils.js [hunk 1]
```diff
@@ -386,18 +386,18 @@ var inChrome = typeof Components != "undefined" && "utils" in Components;
...
+      // Finally, ensure we look good.
+      return {
+        hostname: urlObject.hostname,
+        location: decodeURI(urlObject.href)
+      };
     } catch (ex) {
       if (suppressConsoleError ? !suppressConsoleError : true) {
-        console.error("Error occurred whilst parsing URL:", ex);
+        console.log("Error occurred whilst parsing URL: ", ex);
+        console.trace();
```
Comment (Standard8) [inline shared/js/utils.js:397]: I think we can leave this as console.error - that gives a trace automatically.

File: shared/test/utils_test.js [hunk 2]
```diff
@@ -318,14 +318,17 @@ describe("loop.shared.utils", function() {
         });
     });
 
-    it("should return null if it the url is not valid", function() {
-      expect(sharedUtils.formatURL("hinvalid//url")).eql(null);
+    it("should return null if suppressConsoleError is true and the url is not valid", function() {
+      expect(sharedUtils.formatURL("hinvalid//url", true)).eql(null);
```
Comment (Standard8) [inline shared/test/utils_test.js:322]: No need to pass `true` here. The stub in the before each should prevent issues.

File: shared/test/utils_test.js [hunk 3]
```diff
@@ -318,14 +318,17 @@ describe("loop.shared.utils", function() {
...
     });
 
-    it("should log an error message to the console", function() {
-      sharedUtils.formatURL("hinvalid//url");
+    it("should return null if suppressConsoleError is true and is a malformed URI sequence", function() {
+      expect(sharedUtils.formatURL("http://www.example.com/DNA/pizza/menu/lots-of-different-kinds-of-pizza/%8D%E0%B8%88%E0%B8%A1%E0%B8%A3%E0%8D%E0%B8%88%E0%B8%A1%E0%B8%A3%E0%", true)).eql(null);
```
Comment (Standard8) [inline shared/test/utils_test.js:326]: No need to pass `true` here. The stub in the before each should prevent issues.

**Comment by Standard8** [comment]
I'd suggest making the commit message a little clearer, e.g. `Bug 1240554 - Fix layout bustage in the ui-showcase by providing defaults for the standalone button. Also fix errors from an invalid url.`

File: shared/js/utils.js (hunk 1 above)
Comment (Standard8) [inline shared/js/utils.js:397]: Ok, per dmose on irc, leave it as `console.log`, but lets drop the `console.trace`.

**Comment by Standard8** [comment]
Oh, also, since this isn't actually bug 1240554, I'd just drop the bug number.

r=Standard8.

Don't forget to force-push your rebased branch to here before you --ff-only merge it to master.

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Pull request merged**